lib/filterx/filterx-ref.c

```diff
@@ -195,10 +195,15 @@ _filterx_ref_repr(FilterXObject *s, GString *repr)
 {
   FilterXRef *self = (FilterXRef *) s;
 
-  if (self->value->type->repr)
-    return self->value->type->repr(self->value, repr);
+  return filterx_object_repr(self->value, repr);
```

Comment:
This breaks `append()`, because `filterx_object_repr()` truncates the string.

Reply:
I noticed this in a follow up patch and now fixed it.